cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -368,11 +369,18 @@ public void performAction(HttpRequest request, HttpResponder responder,
     }
 
     if (!isValidAction(action)) {
-      responder.sendStatus(HttpResponseStatus.METHOD_NOT_ALLOWED);
+      responder.sendStatus(HttpResponseStatus.NOT_FOUND);
```

Comment:
I have tried to keep the refactoring here small. The entire class (and some others - ProgramLifecycleService, DefaultStore) have a wild mix of different error handling styles. This needs more refactoring but that would go beyond a bugfix and I don't want to do this so close to a release.See https://issues.cask.co/browse/CDAP-3107